x/ref/runtime/internal/flow/conn/flow.go

```diff
@@ -116,12 +116,9 @@ func (f *flw) DisableFragmentation() {
 // Read and ReadMsg should not be called concurrently with themselves
 // or each other.
 func (f *flw) Read(p []byte) (n int, err error) {
-	f.conn.mu.Lock()
-	ctx := f.ctx
-	f.conn.mu.Unlock()
 	f.markUsed()
-	if n, err = f.q.read(ctx, p); err != nil {
-		f.close(ctx, false, err)
+	if n, err = f.q.read(f.currentContext(), p); err != nil {
```

Comment:
This is fine as a quick fix, but it's also error prone. Can we change Read, ReadMsg and other methods to take a context arg instead of stashing a context in a variable?

Reply:
yes, we can, filed #97 to address this.